include/powsybl/iidm/converter/xml/IidmXmlUtil.hpp

```diff
@@ -192,7 +206,12 @@ class IidmXmlUtil {
     IidmXmlUtil() = delete;
 
 private:
-    static void writeAttributeFromMinimumVersion(const std::string& rootElementName, const std::string& attributeName, bool isNotDefaultValue, const char* errorMessage, const IidmXmlVersion& minVersion, const IidmXmlVersion& contextVersion, const std::function<void()>& write);
+    static void writeAttributeFromMinimumVersion(const std::string& rootElementName, const std::string& attributeName, bool isNotDefaultValue, const char* errorMessage, const IidmXmlVersion& minVersion, const NetworkXmlWriterContext& context, const std::function<void()>& write);
+
+private:
+    static constexpr const char* const MAXIMUM_REASON = "IIDM-XML version should be <= ";
+
+    static constexpr const char* const MINIMUM_REASON = "IIDM-XML version should be >= ";
```

Comment:
Maybe you should move these constants in the ErrorMessage namespace declared at the beginning of this file. In Java the `ErrorMessage` is an enum, so maybe not...  Maybe it could also be hidden in the implementation

Reply:
Ok, as those strings are not supposed to be used from outside `IidmXmlUtil`, I think moving them in the implementation is a good idea => updated